Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -262,10 +274,16 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
         public override async Task<string> GetAccessTokenForRequestAsync(string authUri = null, CancellationToken cancellationToken = default)
         {
             // See: https://developers.google.com/identity/protocols/oauth2/service-account#jwt-auth
-            if (!HasExplicitScopes && authUri != null)
+            if (!HasExplicitScopes && authUri == null)
+            {
+                throw new GoogleApiException(string.Empty, "Invalid OAuth scope or ID token audience provided. A valid authUri and/or OAuth scope is required to proceed.");
+            }
+            // See: https://google.aip.dev/auth/4111
+            if (UseJwtAccessWithScope || !HasExplicitScopes)
             {
                 return await GetOrCreateJwtAccessTokenAsync(authUri).ConfigureAwait(false);
             }
+
             return await base.GetAccessTokenForRequestAsync(authUri, cancellationToken).ConfigureAwait(false);
```

Comment:
This might be just me, but I think it's easier to read if we swapped the if:
```suggestion
            if (HasExplicitScopes && !UseJwtAccessWithScope)
            {
                return await base.GetAccessTokenForRequestAsync(authUri, cancellationToken).ConfigureAwait(false);
            }
            return await GetOrCreateJwtAccessTokenAsync(authUri).ConfigureAwait(false);            
```

Reply:
Indeed, non-jwt case is a narrow case now and it makes sense to describe it as a condition. And SS-JWT as default.